FluentD 1.2.x / Ruby 2.5.1 support

I'm running fluent version 1.2.4 so have upgraded my Ruby version to 2.5.1 to support a number of other fluent plugins.  When attempting to use this plugin on this version, I receive an error that indicates a problem with msgpack.  Can you tell me how to fix, if possible?

Building native extensions. This could take a while...
ERROR:  Error installing fluent-plugin-splunk-ex:
	ERROR: Failed to build gem native extension.

    current directory: /home/dev/.rvm/gems/ruby-2.5.1/gems/msgpack-0.5.12/ext/msgpack
/home/dev/.rvm/rubies/ruby-2.5.1/bin/ruby -r ./siteconf20180817-35088-1iu3y4m.rb extconf.rb
checking for ruby/st.h... yes
checking for st.h... yes
checking for rb_str_replace() in ruby.h... yes
checking for rb_intern_str() in ruby.h... yes
checking for rb_sym2str() in ruby.h... yes
checking for rb_str_intern() in ruby.h... yes
creating Makefile

current directory: /home/dev/.rvm/gems/ruby-2.5.1/gems/msgpack-0.5.12/ext/msgpack
make "DESTDIR=" clean

current directory: /home/dev/.rvm/gems/ruby-2.5.1/gems/msgpack-0.5.12/ext/msgpack
make "DESTDIR="
compiling buffer.c
compiling buffer_class.c
compiling core_ext.c
core_ext.c: In function ‘MessagePack_core_ext_module_init’:
core_ext.c:121:22: error: ‘rb_cFixnum’ undeclared (first use in this function)
     rb_define_method(rb_cFixnum, "to_msgpack", Fixnum_to_msgpack, -1);
                      ^
core_ext.c:121:22: note: each undeclared identifier is reported only once for each function it appears in
core_ext.c:122:22: error: ‘rb_cBignum’ undeclared (first use in this function)
     rb_define_method(rb_cBignum, "to_msgpack", Bignum_to_msgpack, -1);
                      ^
core_ext.c: At top level:
cc1: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-constant-logical-operand" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-parentheses-equality" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-tautological-compare" [enabled by default]
make: *** [core_ext.o] Error 1

--- a/fluent-plugin-splunk-ex.gemspec
+++ b/fluent-plugin-splunk-ex.gemspec
@@ -2,7 +2,7 @@
 
 Gem::Specification.new do |gem|
   gem.name         = "fluent-plugin-splunk-ex"
-  gem.version      = "1.0.1"
+  gem.version      = "1.0.3"
 
   gem.authors      = ["Trevor Gattis"]
   gem.email        = "github@trevorgattis.com"
@@ -17,7 +17,7 @@ Gem::Specification.new do |gem|
   gem.executables = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   gem.require_paths = ['lib']
 
-  gem.add_dependency "fluentd", "~> 0.10.17"
+  gem.add_dependency "fluentd", "~> 0.12.22"
   gem.add_runtime_dependency "json"
 
   gem.add_development_dependency "rake"

